fairscale/utils/testing.py

```diff
@@ -211,3 +212,81 @@ def replacement(*args: Any, **kwargs: Any) -> None:
         return func
 
     return prepare_test
+
+
+class _Block(nn.Module):
+    def __init__(self, embed_dim: int, num_heads: int) -> None:
+        super().__init__()
+        self.ln_1 = nn.LayerNorm(embed_dim)
+        self.ln_2 = nn.LayerNorm(embed_dim)
+        self.attn = nn.MultiheadAttention(embed_dim, num_heads)  # type: ignore
+        self.mlp = nn.Sequential(nn.Linear(embed_dim, embed_dim * 4), nn.GELU(), nn.Linear(embed_dim * 4, embed_dim),)
+
+    def forward(self, *inputs: Any, **kwargs: Any) -> Any:
+        x = inputs[0]
+        attn_mask = torch.full((len(x), len(x)), -float("Inf"), device=x.device, dtype=x.dtype)
+        attn_mask = torch.triu(attn_mask, diagonal=1)
+
+        x = self.ln_1(x)
+        a, _ = self.attn(x, x, x, attn_mask=attn_mask, need_weights=False)
+        x = x + a
+        m = self.mlp(self.ln_2(x))
+        x = x + m
+        return x
+
+
+class GPT2(nn.Module):
```

Comment:
does not take a lot of space, and makes GPT available for testing without any dependency, it's a proven good corner case.